docs/adr/adr_0011.md

```diff
@@ -0,0 +1,167 @@
+<!--
+SPDX-FileCopyrightText: 2021 iteratec GmbH
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+# ADR-0011: Version Numbers
+
+| <!-- -->       | <!-- --> |
+|----------------|----------|
+| **Status**:    | DRAFT |
+| **Date**:      | 2022-01-18 |
+| **Author(s)**: | Max Maass <max.maass@iteratec.com> |
+
+## Context
+Software version numbers should serve as an indicator of compatibility between different versions, both during operation and while updating to a newer release.
```

Comment:
IMHO it would make sense to summarize all our APIs where breaking changes may occur. E.g. the CRDs, the Finding Output Format, the Parser SDK etc. Based on that we can decide which of this API has impact to whom and how. E.g. an Parser SDK API change will not affect an operator of the SCB, but a maintainer of a scanner.

Reply:
I don't quite understand what exactly you propose to do here, so I have not made changes to address this comment. What would be the impact on the versioning from this differentiation? You want to know which kind of change is breaking to whom? For which purpose - being able to give more specific upgrade guidance in the release notes, or should it have a direct impact on the versioning itself?

Reply:
Usually in an architecture documentation you have something like a [component overview](https://docs.arc42.org/section-5/). It is the first level drill down of the whole system. In this component overview each component describes which APIs it provides.

If we would have something like that – not here in the ADR but as general architecture documentation – we'll have an overview to identify all the interfaces of components which we need to cover by this ADR. At the moment we are discussing possible breaking changes here and there, but IMHO we do not share common picture of:

* what components do we have
* what APIs do the use to interface each other
* which of these APIs must be considered for versioning

Reply:
This is wip now.